### PR TITLE
[MNG-6825] Replace Plexus IOUtil with IOUtils from commons-io

### PR DESCRIPTION
**Pull request by timtebeek**
For https://issues.apache.org/jira/browse/MNG-6825

This third PR targets IOUtil, which was not present in apache/maven, hence I've targeted the project that uses it the most.

Regrettably some manual fixes where needed, as the replacement commons-io IOUtils does not silently swallow IOExceptions like Plexus IOUtil did. Hence inlined the old behaviour a couple times and adopted try-with-resources a few other times.

Not entirely happy both with the process for these replacements and with the outcomes. PR opened to gather feedback on whether we would want to target the other replacements similarly, or adopt a different approach, such as more try-with-resources, or no commons-io IOUtils.

### CODE REVIEW

File: doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/AbstractSinkTest.java [hunk 1]
```diff
@@ -110,7 +110,7 @@ public final void testTestDocument() throws IOException {
             SinkTestDocument.generate(testSink);
         } finally {
             testSink.close();
-            IOUtil.close(writer);
+            IOUtils.close(writer);
```
Comment (elharo) [inline doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/AbstractSinkTest.java:113]: This one might be able to use try with resources.

**Review by elharo — APPROVED** [review]

File: doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/AbstractSinkTest.java (hunk 1 above)
Comment (timtebeek) [reply doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/AbstractSinkTest.java:113]: Looked at this briefly, but `Sink` does not implement `AutoCloseable`, which makes it slightly awkward, especially as some of the multiple `Sink#close()` implementations call out to `init()` for instance, so leaving out `testSink.close()` is also not an option. You'd then have to declare `Sink` outside the try with resources, and close that through finally, which could change the order of the `close` calls if I'm not mistaken. Any other suggestions on how to handle this here?

Comment (elharo) [reply doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/AbstractSinkTest.java:113]: https://issues.apache.org/jira/browse/DOXIA-698 but until then this probably works. Does closing the sink close the Writer? If so we don't need IOUtils at all here.

The other pre-existing issue that;s weird in this test is the isXmlSink() branch on line 106. I'm not sure what that's about.

Comment (timtebeek) [reply doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/AbstractSinkTest.java:113]: It's `Sink.close()` documented as that it should:

**Pull request merged**